proto/src/rr/record_data.rs

```diff
@@ -490,12 +490,15 @@ impl RData {
 
         // we should have read rdata_length, but we did not
         let read = decoder.index() - start_idx;
-        if read != rdata_length as usize {
-            return Err(ProtoErrorKind::IncorrectRDataLengthRead {
-                read,
-                len: rdata_length as usize,
-            }.into());
-        }
+        rdata_length
+            .map(|u| u as usize)
+            .verify_unwrap(|rdata_length| read == *rdata_length)
+            .map_err(|rdata_length| {
+                ProtoError::from(ProtoErrorKind::IncorrectRDataLengthRead {
+                    read,
+                    len: rdata_length,
+                })
+            })?;
```

Comment:
I think adopting this construction like elsewhere increases consistency, but there's nothing actually new here.